phoenix-core/src/it/java/org/apache/phoenix/end2end/TruncateTableIT.java

```diff
@@ -0,0 +1,186 @@
+
+        try {
+            conn.createStatement()
+                .execute("CREATE TABLE " + tableName + " (C1 INTEGER PRIMARY KEY)");
+            conn.setAutoCommit(true);
+            PreparedStatement stmt = conn.prepareStatement(
+                "UPSERT INTO " + tableName + " VALUES(1)");
+            stmt.execute();
+            ResultSet rs = conn.createStatement().executeQuery("SELECT COUNT(*) FROM " + tableName);
+            assertTrue(rs.next());
+            assertEquals(1, rs.getInt(1));
+            conn.createStatement().execute("TRUNCATE TABLE " + schemaName + "." + tableName);
+            fail();
+        } catch (SQLException e) {
+            conn.createStatement().execute("DROP TABLE " + tableName);
+        }
+    }
+
+    @Test
+    public void testTruncateTableWithImplicitSchema() throws SQLException {
+        Properties props = new Properties();
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        String schemaName = generateUniqueName();
+        String tableName = generateUniqueName();
+        String createTableWithSchema = "CREATE TABLE " + schemaName + "." + tableName + " (C1 char(2) NOT NULL PRIMARY KEY)";
+
+        try {
+            conn.createStatement().execute(createTableWithSchema);
+            conn.setAutoCommit(true);
+            PreparedStatement stmt = conn.prepareStatement("UPSERT INTO " + schemaName + "." + tableName + " values('a')");
+            stmt.execute();
+            ResultSet rs = conn.createStatement().executeQuery("SELECT COUNT(*) FROM " + schemaName + "." + tableName);
+            assertTrue(rs.next());
+            assertEquals(1, rs.getInt(1));
+            stmt = conn.prepareStatement("UPSERT INTO " + schemaName + "." + tableName + " values('b')");
+            stmt.execute();
+            rs = conn.createStatement().executeQuery("SELECT COUNT(*) FROM " + schemaName + "." + tableName);
+            assertTrue(rs.next());
+            assertEquals(2, rs.getInt(1));
+            conn.close();
+
+            conn = DriverManager.getConnection(getUrl(), props);
+            conn.createStatement().execute("TRUNCATE TABLE " + schemaName + "." + tableName);
+            conn.close();
+
+            conn = DriverManager.getConnection(getUrl(), props);
+            rs = conn.createStatement().executeQuery("SELECT COUNT(*) FROM " + schemaName + "." + tableName);
+            assertTrue(rs.next());
+            assertEquals(0, rs.getInt(1));
+            conn.close();
+
+            conn = DriverManager.getConnection(getUrl(), props);
+            conn.createStatement().execute("DROP TABLE " + schemaName + "." + tableName);
+            conn.close();
+        } catch (SQLException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void testTruncateTableWithExplicitSchema() throws SQLException {
+        Properties props = new Properties();
+        props.setProperty(QueryServices.IS_NAMESPACE_MAPPING_ENABLED, Boolean.toString(true));
```

Comment:
A given Phoenix cluster can only have Namespace mapping enabled or disabled; it can't be enabled for some tables and disabled for others. This usually means either having to shut down the minicluster and spin it back up with namespace enabled, or putting this is in a separate IT test.